qiskit_experiments/library/calibration/half_angle_cal.py

```diff
@@ -50,10 +52,46 @@ def __init__(
             backend: Optional, the backend to run the experiment on.
             schedule_name: The name of the schedule to calibrate which defaults to sx.
             cal_parameter_name: The name of the parameter in the schedule to update. This will
-                default to amp since the complex amplitude contains the phase of the pulse.
+                default to 'angle' in accordance with the naming convention of the
+                'ScalableSymbolicPulse` class.
             auto_update:  Whether or not to automatically update the calibrations. By
                 default this variable is set to True.
+
+        Raises:
+            CalibrationError: if cal_parameter_name is set to `amp`, to reflect the
+                transition from calibrating complex amplitude to calibrating the phase.
+            CalibrationError: if the default cal_parameter_name is used, and it is not
+                a valid parameter of the calibrated schedule.
         """
+        if cal_parameter_name == "amp":
+            raise CalibrationError(
```

Comment:
We can start from deprecation warning with implicit variable replacement.

Reply:
If the user manually specifies `cal_parameter_name == "amp"` it raises the concern that he is using some custom pulse which might not be readily converted to (amp, angle) representation. I think it's safer to raise an error, and let the user verify the issue.

Reply:
That's fair point. This also makes me think we should add metadata for library pulse.